src/gfx/shaders/VolumeMaterial.js

```diff
@@ -91,59 +94,74 @@ void main() {
   volPos = volPos + 0.5;
   volPos.w = 0.5;
 }`,
-    fragmentShader: `\
+      fragmentShader: `\
 varying vec4 volPos;
 void main() {
   gl_FragColor = volPos;
 }`,
-    transparent: false,
-    depthTest: false,
-    depthWrite: false,
-    side: THREE.FrontSide,
-  };
-  return new THREE.ShaderMaterial(settings);
+      transparent: false,
+      depthTest: false,
+      depthWrite: false,
+      side: THREE.FrontSide,
+    };
+    super(settings);
+  }
 }
 
-function FrontFacePosMaterial(params) {
-  const settings = {
-    uniforms: overrideUniforms(params, {}),
-    vertexShader: `\
+class FrontFacePosMaterial extends THREE.ShaderMaterial {
+  constructor(params) {
+    const settings = {
+      uniforms: overrideUniforms(params, {}),
+      vertexShader: `\
 varying vec3 pos;
 void main() {
   // we're assuming local position is in [-0.5, 0.5]
   // we need to offset it to be represented in RGB
   pos = position.xyz + 0.5;
   gl_Position = projectionMatrix * modelViewMatrix * vec4(position, 1.0);
 }`,
-    fragmentShader: `\
+      fragmentShader: `\
 varying vec3 pos;
 void main() {
   gl_FragColor = vec4(pos, 0.5);
 }`,
-    transparent: false,
-    depthTest: false,
-    depthWrite: false,
-    side: THREE.FrontSide,
-  };
-  return new THREE.ShaderMaterial(settings);
+      transparent: false,
+      depthTest: false,
+      depthWrite: false,
+      side: THREE.FrontSide,
+    };
+    super(settings);
+  }
 }
 
-function VolumeMaterial(params) {
-  const settings = {
-    uniforms: overrideUniforms(params, volumeUniforms),
-    vertexShader: `\
+class VolumeMaterial extends THREE.ShaderMaterial {
+  constructor(params) {
+    const settings = {
+      uniforms: overrideUniforms(params, volumeUniforms),
+      vertexShader: `\
 varying vec4 screenSpacePos;
 void main() {
   screenSpacePos = projectionMatrix * modelViewMatrix * vec4(position, 1.0);
   gl_Position = screenSpacePos;
 }`,
-    fragmentShader: volumeFrag,
-    transparent: true,
-    depthTest: true,
-    depthWrite: false,
-    side: THREE.FrontSide,
-  };
-  return new THREE.ShaderMaterial(settings);
+      fragmentShader: volumeFrag,
+      transparent: true,
+      depthTest: true,
+      depthWrite: false,
+      side: THREE.FrontSide,
+    };
+
+    super(settings);
+    this.updateDefines();
+  }
+
+  updateDefines() {
+    const defines = {};
+    defines.ISO_MODE = globalSettings.now.modes.VD.IsoMode;
```

Comment:
This is [marked as a code smell](https://sonarcloud.io/project/issues?id=epam%3Amiew&issues=AWsD4CWFVDQP4Uuf9RPs&open=AWsD4CWFVDQP4Uuf9RPs&pullRequest=162) and it should be easily fixed. Don't define an empty object, create it in the final form.